omigami/models/pls_classifier.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import numpy as np
+from sklearn.cross_decomposition import PLSRegression
+from sklearn.preprocessing import OneHotEncoder
+
+
+def get_vip(model):
+    # Calculate VIP
+    # https://github.com/scikit-learn/scikit-learn/pull/13492
+    # https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3883573/#S6title
+    T = model.x_scores_
+    W = model.x_weights_
+    Q = model.y_loadings_
+    w0, w1 = W.shape
+    s = np.sum(T ** 2, axis=0) * np.sum(Q ** 2, axis=0)
+    s_sum = np.sum(s, axis=0)
+    w_norm = np.array([(W[:, i] / np.linalg.norm(W[:, i])) for i in range(w1)])
+    vip = np.sqrt(w0 * np.sum(s * w_norm.T ** 2, axis=1) / s_sum)
+    return vip
+
+
+class PLSClassifier(PLSRegression):
+    """This class extends the scikit-learn PLS regression to
+    multi-class classification. To do so, it converts the classification problem to
+    a multi dimensional regression. Output classes are one-hot encoded and used
+    to train a PLS regressor. Predictions are given by casting the output
+    of the regression to a binary matrix where the 1s correspond to the maximum
+    of each row. Finally the matrix is converted to classes using the inverse one hot
+    transformation.
+
+    The signaure is analogous to sklearn.cross_decomposition.PLSRegression:
+
+    Args:
```

Comment:
Should we comply with some docstring syntax? We usually use numpy style docs.

Reply:
I'm ok with numpy

Reply:
I'm ok too, I think I was using google by default of my IDE